examples/examples/z_querier.rs

```diff
@@ -0,0 +1,142 @@
+//
+// Copyright (c) 2024 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+use std::time::Duration;
+
+use clap::Parser;
+use zenoh::{key_expr::KeyExpr, query::QueryTarget, Config};
+use zenoh_examples::CommonArgs;
+
+#[tokio::main]
+async fn main() {
+    // initiate logging
+    zenoh::init_log_from_env_or("error");
+
+    let (config, keyexpr, payload, target, timeout) = parse_args();
+
+    println!("Opening session...");
+    let session = zenoh::open(config).await.unwrap();
+
+    println!("Declaring Querier on '{keyexpr}'...");
+    let querier = session
+        .declare_querier(keyexpr)
+        .target(target)
+        .timeout(timeout)
+        .await
+        .unwrap();
+
+    #[cfg(feature = "unstable")]
```

Comment:
`matching_listener` is not available in `z_pub` example, maybe adding it as well there?
Need to check first the feasibility for all the other bindings though in the spirit of keeping all the examples aligned in all the languages.

Reply:
I think it makes sense. Although at least zenoh-pico does not have matching_listener/status implementation. I believe all zenoh-c and zenoh-cpp (if built with zenoh-c) z_pub and z_pub_shm examples already have an option to enable matching listener based on cli arguments.

Reply:
Ok, for rust, should it be also under --add-matching-listener cli flag as it is done today in c/c++ ?

Reply:
Let's keep all the examples consistent across bindings in terms of code flow, logic, and CLI arguments.